hyppo/independence/hhg.py

```diff
@@ -224,3 +309,97 @@ def _pearson_stat(distx, disty):  # pragma: no cover
                     S[i, j] = ((n - 2) * (t12 * t21 - t11 * t22) ** 2) / denom
 
     return S
+
+
+def _point_distance(self, x, y, zx, zy, **kwargs):
+    """
+    For fast HHG,
+    Returns a collection of distances between sample points and chosen centre point
+    """
+    distx = cdist(zx, x, metric=self.compute_distance, **kwargs)
+    disty = cdist(zy, y, metric=self.compute_distance, **kwargs)
+
+    return distx, disty
+
+
+def hoeffdingsD(x, y):
+    """
+    For fast HHG, calculates the Hoeffding's dependence statistic
+    """
+    xin = x
+    yin = y
+    # crop data to the smallest array, length have to be equal
```

Comment:
wait, it really does this? Can you send the link? If it does just use this: https://numpy.org/doc/stable/reference/generated/numpy.resize.html

Reply:
Removed.